File: backend/internal/model/news.go
package model

import "time"

// NewsItem represents a financial news article
type NewsItem struct {
	Title       string    `json:"title"`
	Description string    `json:"description"`
	Source      string    `json:"source"`
	URL         string    `json:"url"`
	ImageURL    string    `json:"image_url,omitempty"`
	PublishedAt time.Time `json:"published_at"`
	Category    string    `json:"category"`
}

File: app/src/api/news.ts
import { fetchAPI } from './base';

export interface NewsItem {
  title: string;
  description: string;
  source: string;
  url: string;
  image_url?: string;
  published_at: string;
  category: string;
}

export interface AINewsSummary {
  date: string;
  summary: string;
  recommendations: string[];
  sentiment: 'positive' | 'negative' | 'neutral' | 'volatile';
  has_breaking_news: boolean;
}

export const news = {
  list: (limit?: number) => fetchAPI<NewsItem[]>(`/news${limit ? `?limit=${limit}` : ''}`),
  summary: () => fetchAPI<AINewsSummary>('/news/summary'),
};
